server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -1392,14 +1392,32 @@ def text(
     ) -> StepContext:
         from pypika.terms import ValueWrapper
 
+        # Since we're using WITH...AS syntax, we add an explicit cast here to provide type
+        # context to the engine. Without that, we might encounter "failed to find conversion
+        # function from "unknown" to text" errors
+        value = step.text
+
+        if isinstance(value, datetime):
+            # ValueWrapper(value) would produce an iso8601 string which
+            # is not properly handled by some backends
+            value_wrapped = ValueWrapper(value.strftime("%Y-%m-%d %H:%M:%S"))
+        else:
+            value_wrapped = ValueWrapper(value)
```

Comment:
I think we should ensure that the object is naive. Maybe in the validator ?Something along the lines of
```python
from zoneinfo import ZoneInfo

if value.tzinfo is not None:
    value = value.astimezone(ZoneInfo('UTC')).replace(tzinfo=None)
```